### PR TITLE
Improve “Encrypting Secret Data at Rest” page

### PR DESCRIPTION
**Pull request by quangthe**
* Add detail example and explain how to configure kube-apiserver static pod
* Minor fix for verify secret command
* Detail example for etcdctl command to get dump secret

### CODE REVIEW

**Comment by sftim** [comment]
/retitle Improve “Encrypting Secret Data at Rest” page

The issue for this is #33431

**Review by sftim — COMMENTED** [review]
Thanks for the PR

@quangthe we avoid putting references to issues in PR titles and commit messages.
Please change this PR so that the commit message:
> kubernetes#33431 docs improvement

is different and GitHub no longer treats that message as meaningful.

/hold
pending that fix.


---

Please read our [style guide](https://kubernetes.io/docs/contribute/style/style-guide/) and aim to follow that. This PR has several aspects that are out of line with that guidance.

**Comment by quangthe** [comment]
Hi @sftim 

Thanks for your feedback. I updated the PR. Could you please review it again?

**Comment by Sea-n** [comment]
Unhold since [the commit message issue](https://github.com/kubernetes/website/pull/35425#pullrequestreview-1052212299) is solved.
/unhold

**Review by sftim — COMMENTED** [review]
Some feedback - I hope this is useful.

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 1]
```diff
@@ -167,13 +210,44 @@ program to retrieve the contents of your Secret.
...
    where `[...]` must be the additional arguments for connecting to the etcd server.
 
+   Example:
+
+   ```
+   ETCDCTL_API=3 etcdctl \
+      --cacert=/etc/kubernetes/pki/etcd/ca.crt \
+      --cert=/etc/kubernetes/pki/etcd/server.crt \
+      --key=/etc/kubernetes/pki/etcd/server.key \
+      get /registry/secrets/default/secret1  | hexdump -C
```
[the remark this answers is not in the document]
Comment (quangthe) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:220]: Thanks @sftim, I updated the PR as your suggestion.

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 2]
```diff
@@ -167,13 +210,49 @@ program to retrieve the contents of your Secret.
...
+   000000c0  d6 1f 37 41 6d 35 ee af  e9 62 a5 35 16 f4 e6 66  |..7Am5...b.5...f|
+   000000d0  08 42 36 fa 14 5b a3 52  f4 cd cb d8 f2 8f 3a c0  |.B6..[.R......:.|
+   000000e0  24 92 7e b9 45 ab 99 4b  f1 4a 18 55 72 04 11 a8  |$.~.E..K.J.Ur...|
+   000000f0  29 92 45 de 18 a0 ef 60  10 4b f8 42 11 e4 87 9a  |).E....`.K.B....|
+   00000100  5a 36 19 10 20 7f 66 80  d6 63 19 dd f0 f3 3a 71  |Z6.. .f..c....:q|
+   00000110  23 3a 0d fc 28 ca 48 2d  6b 2d 46 cc 72 0b 70 4c  |#:..(.H-k-F.r.pL|
+   00000120  a5 fc 35 43 12 4e 60 ef  bf 6f fe cf df 0b ad 1f  |..5C.N`..o......|
+   00000130  82 c4 88 53 02 da 3e 66  ff 0a                    |...S..>f..|
+   0000013a
+   ```
```
Comment (mtardy) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:247]:
````suggestion
   For example:

   ```shell
   ETCDCTL_API=3 etcdctl \
      --cacert=/etc/kubernetes/pki/etcd/ca.crt   \
      --cert=/etc/kubernetes/pki/etcd/server.crt \
      --key=/etc/kubernetes/pki/etcd/server.key  \
      get /registry/secrets/default/secret1 | hexdump -C
   ```

   The output is similar to this (abbreviated):

   ```hexdump
   00000000  2f 72 65 67 69 73 74 72  79 2f 73 65 63 72 65 74  |/registry/secret|
   00000010  73 2f 64 65 66 61 75 6c  74 2f 73 65 63 72 65 74  |s/default/secret|
   00000020  31 0a 6b 38 73 3a 65 6e  63 3a 61 65 73 63 62 63  |1.k8s:enc:aescbc|
   00000030  3a 76 31 3a 6b 65 79 31  3a c7 6c e7 d3 09 bc 06  |:v1:key1:.l.....|
   00000040  25 51 91 e4 e0 6c e5 b1  4d 7a 8b 3d b9 c2 7c 6e  |%Q...l..Mz.=..|n|
   00000050  b4 79 df 05 28 ae 0d 8e  5f 35 13 2c c0 18 99 3e  |.y..(..._5.,...>|
   [...]
   00000110  23 3a 0d fc 28 ca 48 2d  6b 2d 46 cc 72 0b 70 4c  |#:..(.H-k-F.r.pL|
   00000120  a5 fc 35 43 12 4e 60 ef  bf 6f fe cf df 0b ad 1f  |..5C.N`..o......|
   00000130  82 c4 88 53 02 da 3e 66  ff 0a                    |...S..>f..|
   0000013a
   ```
````

**Review by mtardy — COMMENTED** [review]
Hey, I checked your PR, this is awesome! Thanks for taking the time to provide examples over the tutorials, it's truly helpful.

I have suggestions but unfortunately, I cannot use the GitHub suggestion system because I cannot suggest modification over some parts of the file you did not modify, so here it is in raw form. I'm sorry if it's overwhelming as a suggestion but it's mostly moving your paragraphs in the enumeration directly! 😊

In the example in the "Encrypting your data" section, I would incorporate your suggestion directly on the third item of the tutorial:
> 3. Set the `--encryption-provider-config` flag on the `kube-apiserver` to point to
   the location of the config file.

Replacing line `143` to `193` with:
````markdown
1. Set the `--encryption-provider-config` flag on the `kube-apiserver` to point to
   the location of the config file.

   You will need to mount the new encryption config file to the `kube-apiserver` static pod. Here is an example on how to do that:

   1. Save the new encryption config file to `/etc/kubernetes/enc/enc.yaml` on the control-plane node.
   1. Edit the manifest for the `kube-apiserver` static pod: `/etc/kubernetes/manifests/kube-apiserver.yaml` similarly to this:

   ```yaml
   apiVersion: v1
   kind: Pod
   metadata:
     annotations:
       kubeadm.kubernetes.io/kube-apiserver.advertise-address.endpoint: 10.10.30.4:6443
     creationTimestamp: null
     labels:
       component: kube-apiserver
       tier: control-plane
     name: kube-apiserver
     namespace: kube-system
   spec:
     containers:
     - command:
       - kube-apiserver
       ...
       - --encryption-provider-config=/etc/kubernetes/enc/enc.yaml  # <-- add this line
       volumeMounts:
       ...
       - name: enc                           # <-- add this line
         mountPath: /etc/kubernetes/enc      # <-- add this line
         readonly: true                      # <-- add this line
       ...
     volumes:
     ...
     - name: enc                             # <-- add this line
       hostPath:                             # <-- add this line
         path: /etc/kubernetes/enc           # <-- add this line
         type: DirectoryOrCreate             # <-- add this line
     ...
    ```

1. Restart your API server.

{{< caution >}}
Your config file contains keys that can decrypt the contents in etcd, so you must properly restrict
permissions on your control-plane nodes so only the user who runs the `kube-apiserver` can read it.
{{< /caution >}}
````

**Comment by mtardy** [comment]
I can produce a git diff if it's simpler for you! 😊

<details>
  <summary>diff</summary>

````diff
diff --git a/content/en/docs/tasks/administer-cluster/encrypt-data.md b/content/en/docs/tasks/administer-cluster/encrypt-data.md
index 7466ca8e5f..b587cc1ea4 100644
--- a/content/en/docs/tasks/administer-cluster/encrypt-data.md
+++ b/content/en/docs/tasks/administer-cluster/encrypt-data.md
@@ -142,6 +142,45 @@ To create a new Secret, perform the following steps:
 1. Place that value in the `secret` field of the `EncryptionConfiguration` struct.
 1. Set the `--encryption-provider-config` flag on the `kube-apiserver` to point to
    the location of the config file.
+
+   You will need to mount the new encryption config file to the `kube-apiserver` static pod. Here is an example on how to do that:
+
+   1. Save the new encryption config file to `/etc/kubernetes/enc/enc.yaml` on the control-plane node.
+   1. Edit the manifest for the `kube-apiserver` static pod: `/etc/kubernetes/manifests/kube-apiserver.yaml` similarly to this:
+
+   ```yaml
+   apiVersion: v1
+   kind: Pod
+   metadata:
+     annotations:
+       kubeadm.kubernetes.io/kube-apiserver.advertise-address.endpoint: 10.10.30.4:6443
+     creationTimestamp: null
+     labels:
+       component: kube-apiserver
+       tier: control-plane
+     name: kube-apiserver
+     namespace: kube-system
+   spec:
+     containers:
+     - command:
+       - kube-apiserver
+       ...
+       - --encryption-provider-config=/etc/kubernetes/enc/enc.yaml  # <-- add this line
+       volumeMounts:
+       ...
+       - name: enc                           # <-- add this line
+         mountPath: /etc/kubernetes/enc      # <-- add this line
+         readonly: true                      # <-- add this line
+       ...
+     volumes:
+     ...
+     - name: enc                             # <-- add this line
+       hostPath:                             # <-- add this line
+         path: /etc/kubernetes/enc           # <-- add this line
+         type: DirectoryOrCreate             # <-- add this line
+     ...
+   ```
+
 1. Restart your API server.
 
 {{< caution >}}
@@ -149,49 +188,6 @@ Your config file contains keys that can decrypt the contents in etcd, so you mus
 permissions on your control-plane nodes so only the user who runs the `kube-apiserver` can read it.
 {{< /caution >}}
 
-{{< note >}}
-You need to mount the new encryption config file to the kube-apiserver static pod.
-{{< /note >}}
-
-Example steps to set the `--encryption-provider-config` flag on the `kube-apiserver`:
-
-- Save the new encryption config file to `/etc/kubernetes/enc/enc.yaml`.
-
-- Edit the manifest for the `kube-apiserver` static pod: `/etc/kubernetes/manifests/kube-apiserver.yaml`.
-
-  ```yaml
-  apiVersion: v1
-  kind: Pod
-  metadata:
-    annotations:
-      kubeadm.kubernetes.io/kube-apiserver.advertise-address.endpoint: 10.10.30.4:6443
-    creationTimestamp: null
-    labels:
-      component: kube-apiserver
-      tier: control-plane
-    name: kube-apiserver
-    namespace: kube-system
-  spec:
-    containers:
-    - command:
-      - kube-apiserver
-      ...
-      - --encryption-provider-config=/etc/kubernetes/enc/enc.yaml  # <-- add this line
-      volumeMounts:
-      ...
-      - name: enc                           # <-- add this line
-        mountPath: /etc/kubernetes/enc      # <-- add this line
-        readonly: true                      # <-- add this line
-      ...
-    volumes:
-    ...
-    - name: enc                             # <-- add this line
-      hostPath:                             # <-- add this line
-        path: /etc/kubernetes/enc           # <-- add this line
-        type: DirectoryOrCreate             # <-- add this line
-    ...
-  ```
-
 ## Verifying that data is encrypted
 
 Data is encrypted when written to etcd. After restarting your `kube-apiserver`, any newly created or
@@ -210,36 +206,26 @@ program to retrieve the contents of your Secret.
 
    where `[...]` must be the additional arguments for connecting to the etcd server.
 
-   Example:
+   For example:
 
    ```shell
    ETCDCTL_API=3 etcdctl \
-      --cacert=/etc/kubernetes/pki/etcd/ca.crt \
+      --cacert=/etc/kubernetes/pki/etcd/ca.crt   \
       --cert=/etc/kubernetes/pki/etcd/server.crt \
-      --key=/etc/kubernetes/pki/etcd/server.key \
-      get /registry/secrets/default/secret1  | hexdump -C
+      --key=/etc/kubernetes/pki/etcd/server.key  \
+      get /registry/secrets/default/secret1 | hexdump -C
    ```
 
-   the sample output:
+   The output is similar to this (abbreviated):
 
-   ```
+   ```hexdump
    00000000  2f 72 65 67 69 73 74 72  79 2f 73 65 63 72 65 74  |/registry/secret|
    00000010  73 2f 64 65 66 61 75 6c  74 2f 73 65 63 72 65 74  |s/default/secret|
    00000020  31 0a 6b 38 73 3a 65 6e  63 3a 61 65 73 63 62 63  |1.k8s:enc:aescbc|
    00000030  3a 76 31 3a 6b 65 79 31  3a c7 6c e7 d3 09 bc 06  |:v1:key1:.l.....|
    00000040  25 51 91 e4 e0 6c e5 b1  4d 7a 8b 3d b9 c2 7c 6e  |%Q...l..Mz.=..|n|
    00000050  b4 79 df 05 28 ae 0d 8e  5f 35 13 2c c0 18 99 3e  |.y..(..._5.,...>|
-   00000060  44 ae f4 d3 ea f4 40 d1  58 b4 1c 33 7c ba c7 71  |D.....@.X..3|..q|
-   00000070  eb ea 34 f0 2f a3 5a 84  52 96 7c 3f c8 ee a9 50  |..4./.Z.R.|?...P|
-   00000080  bc 83 4f 8b 8e 62 ab 90  c9 eb 41 ca 9b 6d fc 93  |..O..b....A..m..|
-   00000090  7c cf ec 04 0f 99 0c 55  f0 7f da d8 80 82 2a 5f  ||......U......*_|
-   000000a0  be 9c ed 40 0f 3b 6a 41  46 13 8d 2a b3 79 29 83  |...@.;jAF..*.y).|
-   000000b0  57 22 cf 92 a6 d6 92 58  ea 65 51 17 a3 24 be f0  |W".....X.eQ..$..|
-   000000c0  d6 1f 37 41 6d 35 ee af  e9 62 a5 35 16 f4 e6 66  |..7Am5...b.5...f|
-   000000d0  08 42 36 fa 14 5b a3 52  f4 cd cb d8 f2 8f 3a c0  |.B6..[.R......:.|
-   000000e0  24 92 7e b9 45 ab 99 4b  f1 4a 18 55 72 04 11 a8  |$.~.E..K.J.Ur...|
-   000000f0  29 92 45 de 18 a0 ef 60  10 4b f8 42 11 e4 87 9a  |).E....`.K.B....|
-   00000100  5a 36 19 10 20 7f 66 80  d6 63 19 dd f0 f3 3a 71  |Z6.. .f..c....:q|
+   [...]
    00000110  23 3a 0d fc 28 ca 48 2d  6b 2d 46 cc 72 0b 70 4c  |#:..(.H-k-F.r.pL|
    00000120  a5 fc 35 43 12 4e 60 ef  bf 6f fe cf df 0b ad 1f  |..5C.N`..o......|
    00000130  82 c4 88 53 02 da 3e 66  ff 0a                    |...S..>f..|
````

</details>

File: content/en/docs/tasks/administer-cluster/encrypt-data.md (hunk 2 above)
Comment (quangthe) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:247]: Thanks @mtardy for your helpful suggestion!

Comment (quangthe) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md]: The PR updated with the suggestions.

**Comment by mtardy** [comment]
Glad you liked it! :)

For the reviewers, [preview here](https://deploy-preview-35425--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/encrypt-data/) (btw you can edit your initial message with that to make it easier for reviewers to look into your modifications).

/lgtm

**Comment by sftim** [comment]
Thanks for opening this PR. It's very welcome.

File: content/en/docs/tasks/administer-cluster/encrypt-data.md [hunk 3]
```diff
@@ -142,6 +142,45 @@ To create a new Secret, perform the following steps:
 1. Place that value in the `secret` field of the `EncryptionConfiguration` struct.
 1. Set the `--encryption-provider-config` flag on the `kube-apiserver` to point to
    the location of the config file.
+
+   You will need to mount the new encryption config file to the `kube-apiserver` static pod. Here is an example on how to do that:
```
Comment (jimangel) [inline content/en/docs/tasks/administer-cluster/encrypt-data.md:146]:
```suggestion
   Mount the new encryption config file to the `kube-apiserver` static pod.
```

Comment (jimangel) [reply content/en/docs/tasks/administer-cluster/encrypt-data.md:146]: The main takeaway is to remove the word `will` based on k/website's [style-guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-present-tense) requiring present tense. The other parts of the suggestion are non-blocking.

**Review by jimangel — COMMENTED** [review]
Minor nits.
/lgtm

**Comment by Sea-n** [comment]
Hey @quangthe,

It's been a while since the last update, would you like to address issues in [the previous comment](https://github.com/kubernetes/website/pull/35425#discussion_r965059431) so that we can move forward?

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-35425--kubernetes-io-main-staging.netlify.app/docs/tasks/administer-cluster/encrypt-data/#encrypting-your-data

Thanks @quangthe .
/approve

**Pull request merged**